standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/DataConnectorProviderFactory.java

```diff
@@ -89,19 +110,23 @@ public static synchronized IDataConnectorProvider getDataConnectorProvider(Datab
     default:
       throw new MetaException("Data connector of type " + connector.getType() + " not implemented yet");
     }
-    cache.put(connector.getName().toLowerCase(), provider);
+    dataConnectorCache.put(connector.getName().toLowerCase(), provider);
     return provider;
   }
 
-  public void shutdown() {
-    for (IDataConnectorProvider provider: cache.values()) {
-      try {
-        provider.close();
-      } catch(Exception e) {
-        LOG.warn("Exception invoking close on dataconnectorprovider:" + provider, e);
-      } finally {
-        cache.clear();
+  /**
+   * After executing Drop or Alter DDL on a dataConnector, we should update cache to clean the dataConnector
+   * to avoid using the invalid dataConnector next time.
+   * @param dcName dataConnector to be cleaned
+   */
+  public static synchronized void updateDataConnectorCache(String dcName) {
```

Comment:
As this method is serving specific case of case of invalidation of cache, can this be renamed to invaidateDataConnectorFromCache?

Reply:
Sure. Renamed.